app/src/main/java/org/mozilla/fenix/settings/creditcards/String.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.creditcards
+
+private const val MAX_CREDIT_CARD_NUMBER_LENGTH = 19
+private const val MIN_CREDIT_CARD_NUMBER_LENGTH = 12
+
+/**
+ * Strips characters other than digits from a string.
+ * Used to strip a credit card number user input of spaces and separators.
+ */
+
+fun String.toCreditCardNumber(): String {
+    return this.filter { it.isDigit() }
+}
+
+/**
+ * Uses string size and Luhn Algorithm check (https://en.wikipedia.org/wiki/Luhn_algorithm)
+ * to validate a credit card number
+ */
+@Suppress("MagicNumber")
+fun String.validateCreditCardNumber(): Boolean {
```

Comment:
Nice, this saves me the trouble of implementing the Luhn Algorithm. Just an FYI this will probably be refactored in some capacity when I finish up the credit card validation that will return the credit card issuer based on the card number in https://github.com/mozilla-mobile/android-components/issues/9813.

One immediate benefit of having this for now is that we can probably use this for the form validation in the meantime in https://github.com/mozilla-mobile/fenix/issues/18266.